### PR TITLE
nimmi89/c01-aws02

### PR DESCRIPTION
**Pull request by nimmi89**
# Description

Exercise : C01-AWS02

Student Name: Nimmi Varghese

> Creation of Instance roles to access AWS S3 Resources

## Type of change

- [ ] Exercise Submission (one exercise per PR)
  - [ ] This PR is just for **one** exercise of a class
  - [ ] PR name follows the pattern `<github-username>/<exercise-number>`
  - [ ] Added exercise files inside folder `/classes/<class- number>/exercises/<exercise-number>/<github-username>/`
	- [ ] Review [exercise submission steps](./README.md#Exercises)
- [ ] Class content (instructors/admins)
- [ ] Documentation update
- [ ] Repo management

### CODE REVIEW

File: classes/01class/exercises/c01-aws02/nimmi89/README.md [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+       _|  (     /   Amazon Linux 2 AMI
+      ___|\___|___|
+
+https://aws.amazon.com/amazon-linux-2/
+4 package(s) needed for security, out of 8 available
+Run "sudo yum update" to apply all updates.
+[ec2-user@ip-172-31-32-188 ~]$ aws s3 ls
+2020-06-17 08:16:40 c01-aws02-bucket
+
+S3 Access Successful in the EC2 using IAM Roles!!!!!!!!!!!
```
Comment (caiocezart) [inline classes/01class/exercises/c01-aws02/nimmi89/README.md:144]: well done!!!

Comment (nimmi89) [reply classes/01class/exercises/c01-aws02/nimmi89/README.md:144]: To upload a file from instance into s3, s3 full access[aws managed policy] will give unnecessary privileges which are not good for security. So, create a customer-managed policy which allows to put object only and attach it to the role.

File: classes/01class/exercises/c01-aws02/nimmi89/README.md [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+
+>>>Create a policy to upload object onto bucket[Giving only the required privileges]
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Sid": "VisualEditor0",
+            "Effect": "Allow",
+            "Action": "s3:PutObject",
+            "Resource": "*"
```
Comment (kikobr82) [inline classes/01class/exercises/c01-aws02/nimmi89/README.md:174]: you're giving the required privilege, but you're allowing this privilege to any bucket.
Would be better to restrict the bucket that this role will have access to. Can you make this improvement?

Comment (nimmi89) [reply classes/01class/exercises/c01-aws02/nimmi89/README.md]: Yes, I thought so initially. But then I thought I don't have many buckets anyway. Makes Perfect sense though!have made the changes!

**Pull request merged**